### PR TITLE
TASK: Refactor absolute namespaces in Flow / Resource Classes

### PR DESCRIPTION
**Pull request by daniellienert**
Also refactored legacy array syntax.

### CODE REVIEW

**Comment by hhoechtl** [comment]
I just read through it and it looks good. Just one question @daniellienert, did you start the renaming of PHP 7 protected words on purpose, because you just did it in `ResourceRepository` but not in the other classes?

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php [hunk 1]
```diff
@@ -46,7 +47,7 @@ public function publishCollection(CollectionInterface $collection)
      * @param resource $sourceStream Stream of the source to publish
      * @param string $relativeTargetPathAndFilename relative path and filename in the target directory
      * @throws Exception
-     * @throws \TYPO3\Flow\Utility\Exception
+     * @throws Exception
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php]: see the issue with `Exception` imports? ;) Use `Utility\Exception` here

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemTarget.php [hunk 2]
```diff
@@ -169,14 +175,14 @@ public function getName()
...
-     * @param \TYPO3\Flow\Resource\CollectionInterface $collection The collection to publish
+     * @param CollectionInterface $collection The collection to publish
      * @return void
      * @throws Exception
      */
     public function publishCollection(CollectionInterface $collection)
     {
         foreach ($collection->getObjects() as $object) {
-            /** @var \TYPO3\Flow\Resource\Storage\Object $object */
+            /** @var Object $object */
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemTarget.php]: `Object` is another candidate for ambiguous import name, though there is no "Object" class in PHP, but only the "object" type

**Comment by albe** [comment]
Two more little nitpicks and I'm with hhoechtl, we should already use senseful imports for `Resource` everywhere here. `FlowResource` or maybe even `PersistentResource` to already use the name we will use later on (https://discuss.neos.io/t/renaming-flow-core-classes-resource-and-object/1200) - in that case also use `StorageObject` for `Object`.

**Comment by daniellienert** [comment]
Thanks @albe / @hhoechtl - Clearified Exception naming, changed Object / Resource to discussed names.

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourcePointer.php [hunk 3]
```diff
@@ -19,7 +19,7 @@
  * This class is deprecated. Please simply use the Resource->getHash() method instead.
  *
  * @deprecated
- * @see \TYPO3\Flow\Resource\Resource
+ * @see Resource
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourcePointer.php]: PersistentResource

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceTypeConverter.php [hunk 4]
```diff
@@ -78,12 +80,12 @@ class ResourceTypeConverter extends AbstractTypeConverter
...
      * @var array<string>
      */
-    protected $sourceTypes = array('string', 'array');
+    protected $sourceTypes = ['string', 'array'];
 
     /**
      * @var string
      */
-    protected $targetType = 'TYPO3\Flow\Resource\Resource';
+    protected $targetType = Resource::class;
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceTypeConverter.php]: PersistentResource

**Comment by daniellienert** [comment]
@albe Thanks again. I  hope I've got them all now. Would you please have another look?

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php [hunk 5]
```diff
@@ -45,27 +46,26 @@ public function publishCollection(CollectionInterface $collection)
      *
      * @param resource $sourceStream Stream of the source to publish
      * @param string $relativeTargetPathAndFilename relative path and filename in the target directory
-     * @throws Exception
-     * @throws \TYPO3\Flow\Utility\Exception
+     * @throws TargetException
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php]: Does this really no longer throw an `Utility\Exception`? I think `createDirectoryRecursively` still could throw that

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php [hunk 6]
```diff
@@ -415,7 +418,7 @@ public function getCollection($collectionName)
     /**
      * Returns an array of currently known Collection instances
      *
-     * @return array<\TYPO3\Flow\Resource\CollectionInterface>
+     * @return array<PersistentResource\CollectionInterface>
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php]: I think that one overdid the name refactoring :D

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php [hunk 7]
```diff
@@ -426,11 +429,11 @@ public function getCollections()
      * Returns an array of Collection instances which use the given storage
      *
      * @param StorageInterface $storage
-     * @return array<\TYPO3\Flow\Resource\CollectionInterface>
+     * @return array<PersistentResource\CollectionInterface>
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php]: Same as above ;)

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php [hunk 8]
```diff
@@ -115,14 +118,14 @@ public function initialize()
     /**
      * Imports a resource (file) from the given location as a persistent resource.
      *
-     * On a successful import this method returns a Resource object representing the
+     * On a successful import this method returns a PersistentResource object representing the
      * newly imported persistent resource and automatically publishes it to the configured
      * publication target.
      *
-     * @param string | resource $source A URI (can therefore also be a path and filename) or a PHP resource stream(!) pointing to the Resource to import
+     * @param string | PersistentResource $source A URI (can therefore also be a path and filename) or a PHP resource stream(!) pointing to the Resource to import
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/ResourceManager.php]: I think that is really supposed to be PHP 'resource', not a Flow PersistentResource

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/CollectionInterface.php [hunk 9]
```diff
@@ -88,8 +91,8 @@ public function getObjects();
      * Returns a stream handle of the given persistent resource which allows for opening / copying the resource's
      * data. Note that this stream handle may only be used read-only.
      *
-     * @param Resource $resource The resource to retrieve the stream for
-     * @return resource | boolean The resource stream or FALSE if the stream could not be obtained
+     * @param PersistentResource $resource The resource to retrieve the stream for
+     * @return PersistentResource | boolean The resource stream or FALSE if the stream could not be obtained
```
Comment (albe) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/CollectionInterface.php]: returns a PHP 'resource' actually

**Comment by albe** [comment]
Unfortunately now there are some over-doings ;) But if those are resolved it's ok from my side

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php (hunk 5 above)
Comment (daniellienert) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Target/FileSystemSymlinkTarget.php]: Correct - but I think don't annotate nested methods exceptions anymore

**Comment by daniellienert** [comment]
And again, thanks for the review @albe!

**Comment by albe** [comment]
This time you get a :heart: from me, thx for the tremendous effort! :)

**Pull request merged**